gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/nfp_node_driver.py

```diff
@@ -376,8 +380,18 @@ def create(self, context):
         self._set_node_instance_network_function_map(
             context.plugin_session, context.current_node['id'],
             context.instance['id'], network_function_id)
-        self._wait_for_network_function_operation_completion(
-            context, network_function_id, operation='create')
+
+        # Check for NF status in a separate thread
+        gth = self.thread_pool.spawn(self._wait_for_network_function_operation_completion,
+                context, network_function_id, operation='create')
+
+        self.active_threads.append(gth)
+
+        # At last wait for the threads to complete, success/failure/timeout
+        if len(self.active_threads) == self.sc_node_count:
+            for gth in self.active_threads:
+                gth.wait()
```

Comment:
The thread function is throwing an exception if a network function goes to ERROR state. In such case, here error is not percolated to GBP to detect chain create failure. GBP thinks that chain create is successful even if one node create fails.